guides/release/templates/handlebars-basics.md

```diff
@@ -1,85 +1,203 @@
-if the value of one of those properties changes, the DOM will be updated
-automatically.
+If you use JavaScript to change the values of `firstName` or `lastName` in the Controller, what the user sees will be automatically updated! In Ember, this is referred to as "data binding."
 
-As an application grows in size, it will have many templates backed by
-controllers and components.
+Here's another example using a Component.
+Components are usually made of two files, a JavaScript file `my-component.js` and a
+template file with the same name, `my-component.js`. Whatever attributes are defined
+in the Component's JavaScript are available for use in the Template:
 
-### Helpers
+```javascript {data-filename=app/components/my-component.js}
+import Component from '@ember/component';
 
-Ember Helpers are functions that can compute values and can be used in any template.
+export default Component.extend({
+  firstName: 'Jessica',
+  lastName: 'Jordan'
+});
+```
 
-Ember gives you the ability to [write your own helpers](../writing-helpers/), to bring a minimum of logic into Ember templating.
+```handlebars {data-filename=app/templates/components/my-component.hbs}
+Hello, <strong>{{this.firstName}} {{this.lastName}}</strong>!
+```
+
+This would render:
+
+```html
+Hello, <strong>Jessica Jordan</strong>!
+```
+
+## Things you might see in a Template
+
+A lot more than just HTML markup can go in Templates. In the other pages of this guide, we will cover the features one at a time. In general, special Ember functionality will appear inside curly braces, like this: `{{example}}`. Here are a few examples of Ember Handlebars in action:
+
+Route example:
+```hbs {data-filename=app/templates/application.hbs data-update=true}
+{{!-- Example of a comment that will be invisible --}}
+
+{{!-- outlet determines where a child route's content should render. Don't delete it until you know more about it! --}}
+<div>
+  {{outlet}}
+</div>
+
+{{!-- One way to use a component within a template. This is called "Angle Bracket Component syntax." --}}
+<MyComponent />
+
+```
+
+Component example:
+```hbs {data-filename=app/components/templates/my-component.hbs data-update=true}
+{{!-- An attribute that is defined in a component's JavaScript file --}}
+{{this.numberOfSquirrels}}
+
+{{!-- Some data passed down from a parent component or controller --}}
+{{weatherStatus}}
+
+{{!-- This button uses Ember Actions to make it interactive. A method named `plantATree` is called when the button is clicked --}}
+<button onclick={{action 'plantATree'}}>
+  More trees!
+<button>
+
+{{!-- Here's an example of template logic in action. If the `this.skyIsBlue` attribute is `true`, the text inside will be shown --}}
+{{#if this.skyIsBlue}}
+  If the skyIsBlue attribute is true, show this message
+{{/if}}
+
+{{!-- You can pass a whole block of markup and handlebars content from one component to another. yield is where the block shows up when the page is rendered --}}
+{{yield}}
+```
```

Comment:
personally it feels a bit overwhelming with all the template comments.
comments in hbs are `{{! }}`, are you using `{{!-- --}}` for similarity with HTML?

Reply:
I personally usually use `<!-- -->` since folks recognize it as an HTML comment, and syntax highlighters seem to recognize it more often. I realize that Handlebars comments have the benefit of not being rendered supposedly, but I imagine that standard comments could be removed as well if we wanted?

Reply:
TIL. I always used `{{!-- --}}` and didn't know I could do differently!

Swapped all but one for `<!-- -->`